src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -74,13 +74,13 @@ private void ParseRuleAttributes(XElement rule, UrlRewriteRuleBuilder builder)
             var match = rule.Element(RewriteTags.Match);
             if (match == null)
             {
-                ThrowUrlFormatException(rule, "Cannot have rule without match");
+                throw new UrlRewriteParseException(rule, "Cannot have rule without match");
```

Comment:
I realize that this wasn't you but while you're introducing changes can you make these exception messages more consistent. So this would be "Condition must have an associated match" etc